landlab/components/network_sediment_transporter/bed_parcel_initializers.py

```diff
@@ -0,0 +1,782 @@
+"""
+Landlab components to initialize river bed sediment "parcels", represented as
+items in a landlab :class:`~.DataRecord`, in each link in a river network (represented by
+a landlab :class:`~.NetworkModelGrid`). The different *BedParcelInitializers* allow the user
+to define the median grain size on a given link several different ways.
+
+.. codeauthor:: Eric Hutton, Allison Pfeiffer, Muneer Ahammad, and Jon Czuba
+"""
+
+import warnings
+
+import numpy as np
+import scipy.constants
+from landlab import Component
+from landlab.data_record import DataRecord
+from landlab.grid.network import NetworkModelGrid
+
+_OUT_OF_NETWORK = -2
+
+
+class BedParcelInitializerBase(Component):
+    def __init__(
+        self,
+        grid,
+        time=[0.0],
```

Comment:
@pfeiffea, @keckje Why is the *time* keyword a `list` rather than a `float`? Can we change this to `time=0.0` and then convert it to a list later on with `self._time = [time]`?

It's not a good idea to use mutable types (like lists) for keywords, so if we can't use a `float`, how about a tuple (i.e. `time=(0.0,)`)?

Reply:
@mcflugen fixed. Now a float.

Reply:
Thanks!